python/ray/data/tests/test_dataset_formats.py

```diff
@@ -1457,6 +1457,30 @@ def test_csv_write_block_path_provider(
     assert df.equals(ds_df)
 
 
+def test_tensorflow_datasource(ray_start_regular_shared):
+    import tensorflow as tf
+    import tensorflow_datasets as tfds
+
+    tf_dataset = tfds.load("mnist", split=["train"], as_supervised=True)[0]
+    expected_data = list(tf_dataset)
+
+    def dataset_factory():
+        return tfds.load("mnist", split=["train"], as_supervised=True)[0]
+
+    ray_dataset = ray.data.read_datasource(
+        SimpleTensorFlowDatasource(), parallelism=1, dataset_factory=dataset_factory
+    ).fully_executed()
+
+    assert ray_dataset.num_blocks() == 1
+
+    actual_data = ray_dataset.take_all()
+    for (expected_features, expected_label), (actual_features, actual_label) in zip(
+        expected_data, actual_data
+    ):
+        tf.debugging.assert_equal(expected_features, actual_features)
+        tf.debugging.assert_equal(expected_label, actual_label)
```

Comment:
Could use suggestions on how to simplify this part of the test. I'm not super familiar with TensorFlow, so I figure there's a better API to use here.

Reply:
This looks fine to me.